hibernate-reactive-core/src/test/java/org/hibernate/reactive/containers/H2Database.java

```diff
@@ -0,0 +1,122 @@
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.UUID;
+
+import org.hibernate.type.NumericBooleanType;
+import org.hibernate.type.TextType;
+import org.hibernate.type.TrueFalseType;
+import org.hibernate.type.YesNoType;
+import org.hibernate.type.descriptor.java.PrimitiveByteArrayTypeDescriptor;
+
+public class H2Database implements TestableDatabase {
+	public static H2Database INSTANCE = new H2Database();
+
+	private static Map<Class<?>, String> expectedDBTypeForClass = new HashMap<>();
+
+	static {{
+		expectedDBTypeForClass.put( boolean.class, "BOOLEAN" );
+		expectedDBTypeForClass.put( Boolean.class, "BOOLEAN" );
+		expectedDBTypeForClass.put( NumericBooleanType.class, "INTEGER" );
+		expectedDBTypeForClass.put( TrueFalseType.class, "CHARACTER" );
+		expectedDBTypeForClass.put( YesNoType.class, "CHARACTER" );
+		expectedDBTypeForClass.put( int.class, "INTEGER" );
+		expectedDBTypeForClass.put( Integer.class, "INTEGER" );
+		expectedDBTypeForClass.put( long.class, "BIGINT" );
+		expectedDBTypeForClass.put( Long.class, "BIGINT" );
+		expectedDBTypeForClass.put( float.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( Float.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( double.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( Double.class, "DOUBLE PRECISION" );
+		expectedDBTypeForClass.put( byte.class, "TINYINT" );
+		expectedDBTypeForClass.put( Byte.class, "TINYINT" );
+		expectedDBTypeForClass.put( PrimitiveByteArrayTypeDescriptor.class, "BINARY VARYING" );
+		expectedDBTypeForClass.put( URL.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( TimeZone.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( Date.class, "DATE" );
+		expectedDBTypeForClass.put( Timestamp.class, "TIMESTAMP" );
+		expectedDBTypeForClass.put( Time.class, "TIME" );
+		expectedDBTypeForClass.put( LocalDate.class, "DATE" );
+		expectedDBTypeForClass.put( LocalTime.class, "time" );
+		expectedDBTypeForClass.put( LocalDateTime.class, "TIMESTAMP" );
+		expectedDBTypeForClass.put( BigInteger.class, "NUMERIC" );
+		expectedDBTypeForClass.put( BigDecimal.class, "NUMERIC" );
+		expectedDBTypeForClass.put( Serializable.class, "BINARY VARYING" );
+		expectedDBTypeForClass.put( UUID.class, "binary" );
+		expectedDBTypeForClass.put( Instant.class, "datetime" );
+		expectedDBTypeForClass.put( Duration.class, "bigint" );
+		expectedDBTypeForClass.put( Character.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( char.class, "VARCHAR_IGNORECASE" );
+		expectedDBTypeForClass.put( TextType.class, "text" );
+		expectedDBTypeForClass.put( String.class, "VARCHAR_IGNORECASE" );
+	}}
+
+	private String getRegularJdbcUrl() {
+		return "jdbc:h2:~/test;DATABASE_TO_UPPER=FALSE";
+	}
+
+	@Override
+	public String getJdbcUrl() {
+		return "jdbc:h2:~/test;DATABASE_TO_UPPER=FALSE";
```

Comment:
The dialect we pick is not going to work if the user doesn't have this property in the url.
I think, we should add it as a default if it's missing.